src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -180,4 +180,14 @@ public void delete(Long id) throws NoSuchElementException {
             lock.writeLock().unlock();
         }
     }
+
+    @Override
+    public void updateQuantity(Long id, int quantity) {
+        lock.writeLock().lock();
+        try {
+            getProduct(id).setStock(getProduct(id).getStock() - quantity);
```

Comment:
Add variable to getProduct by id and reuse it so the method will not be called twice